kopf/structs/containers.py

```diff
@@ -13,9 +13,33 @@
 from kopf.structs import bodies
 
 
+class ObjectDict(dict):
+    """ A container to hold arbitrary keys-fields assigned by the users. """
+
+    def __setattr__(self, key, value):
+        self[key] = value
+
+    def __delitem__(self, key):
+        try:
+            del self[key]
+        except KeyError as e:
+            raise AttributeError(str(e))
```

Comment:
Doesn't `del self[key]` call `self.__delitem__(key)`? Could it be that you intended to implement [`__delattr__`](https://docs.python.org/3/reference/datamodel.html#object.__delattr__)? If not, I do think that implementing the latter would be a good idea for consistency reasons.

Reply:
Oops. That was `__delattr__`, of course. Fixed. And few more tests added for deletion of the attrs & keys.